blueoil/common.py

```diff
@@ -57,3 +60,45 @@ def get_color_map(length):
     # This function generate arbitrary length color map.
     color_map = COLOR_MAP * int(math.ceil(length / len(COLOR_MAP)))
     return color_map[:length]
+
+
+# For replacing the Matplotlib Jet colormap, we use the Turbo color map
+# https://ai.googleblog.com/2019/08/turbo-improved-rainbow-colormap-for.html
+# The colormap allows for a large number of quantization levels:
+# https://tinyurl.com/ybm3kpql
+
+# Referred from the following gist:
+# https://gist.github.com/mikhailov-work/ee72ba4191942acecc03fe6da94fc73f
+# Copyright 2019 Google LLC.
+# SPDX-License-Identifier: Apache-2.0
+
+# Changes:
+# 1. Vectorized the implementation using numpy
+# 2. Use numpy.modf to get integer and float parts
+# 3. Provided an example in comments
+
+def apply_color_map(image):
+    turbo_cmap_data = np.asarray(TURBO_CMAP_DATA)
+    x = np.asarray(image)
+    x = x.clip(0., 1.)
+
+    # Use numpy.modf to get the integer and decimal parts of feature values
+    # in the input feature map (or heatmap) that has to be colored.
+    # Example:
+    #   >>> import numpy as np
+    #   >>> x = np.array([1.2, 2.3, 4.5, 20.45, 6.75, 8.88])
+    #   >>> f, i = np.modf(x)       # returns a tuple of length 2
+    #   >>> print(i.shape, f.shape)
+    #   >>> (6,) (6,)
+    #   >>> print(i)
+    #   >>> [ 1.  2.  4. 20.  6.  8.]
+    #   >>> print(f)
+    #   >>> [0.2  0.3  0.5  0.45 0.75 0.88]
```

Comment:
some lines wouldn't have `>>>` on the interpreter.

Reply:
Got it. Will update that.